### PR TITLE
Integrates frontend and backend of Confirm Case Details page

### PR DESCRIPTION
**Pull request by amprokop**
Adds backend calls to Certification V2 and refactors lots of Redux.

- persists ConfirmCaseDetails fields to the DB
- refactors Redux ConfirmCaseDetails actions into their own file
- refactors ConfirmCaseDetails reducers into their own file

Testing:
db_persistence

We *still* have no unit tests here. This is gonna be a problem, but isn't so bad since our launch date is still awhile out. I'll write unit tests for this file next.

### CODE REVIEW

File: app/models/certification.rb [hunk 1]
```diff
@@ -27,12 +18,27 @@ def start!
       soc_matching_at:     calculate_soc_matching_at,
       ssocs_required:      calculate_ssocs_required,
       ssocs_matching_at:   calculcate_ssocs_matching_at,
-      form8_started_at:    (certification_status == :started) ? now : nil
+      form8_started_at:    (certification_status == :started) ? now : nil,
+      vacols_hearing_preference: appeal.hearing_request_type,
```
Comment (amprokop) [inline app/models/certification.rb]: we're saving these two fields just for analytics - they won't be user updateable

File: client/app/certification/actions/ConfirmCaseDetails.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+export const changeRepresentativeType = (representativeType) => ({
+  type: Constants.CHANGE_REPRESENTATIVE_TYPE,
+  payload: {
+    representativeType
+  }
+});
+
+export const changeOtherRepresentativeType = (otherRepresentativeType) => ({
+  type: Constants.UPDATE_PROGRESS_BAR,
```
Comment (sharonwarner) [inline client/app/certification/actions/ConfirmCaseDetails.js]: Should be CHANGE_OTHER_REPRESENTATIVE_TYPE

File: client/app/certification/actions/ConfirmCaseDetails.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  payload: {
+    otherRepresentativeType
+  }
+});
+
+export const failValidation = (invalidFields) => ({
```
Comment (sharonwarner) [inline client/app/certification/actions/ConfirmCaseDetails.js]: I think the names of the functions should match. So either FAILED_VALIDATION and failedValidation or FAIL_VALIDATION and failValidation.

Comment (amprokop) [inline client/app/certification/actions/ConfirmCaseDetails.js]: good point will change

**Pull request merged**